flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateDownloader.java

```diff
@@ -36,43 +36,47 @@
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-
-import static org.apache.flink.runtime.concurrent.Executors.newDirectExecutorService;
 
 /**
- * Data transfer utils for {@link RocksDBKeyedStateBackend}.
+ * Help class for downloading RocksDBState.
  */
-class RocksDbStateDataTransfer {
+public class RocksDBStateDownloader extends RocksDBStateDataTransfer {
+	public RocksDBStateDownloader(int restoringThreadNum) {
+		super(restoringThreadNum);
+	}
 
-	static void transferAllStateDataToDirectory(
+	/**
+	 * Transfer all state data to the target directory using specified number of threads.
+	 *
+	 * @param restoreStateHandle Handles used to retrieve the state data.
+	 * @param dest The target directory which the state data will be stored.
+	 * @param closeableRegistry Which all the inputStream/outputStream will be registered and unregistered.
+	 *
+	 * @throws Exception Thrown if can not transfer all the state data.
+	 */
+	public void transferAllStateDataToDirectory(
 		IncrementalKeyedStateHandle restoreStateHandle,
 		Path dest,
-		int restoringThreadNum,
 		CloseableRegistry closeableRegistry) throws Exception {
 
 		final Map<StateHandleID, StreamStateHandle> sstFiles =
 			restoreStateHandle.getSharedState();
 		final Map<StateHandleID, StreamStateHandle> miscFiles =
 			restoreStateHandle.getPrivateState();
 
-		downloadDataForAllStateHandles(sstFiles, dest, restoringThreadNum, closeableRegistry);
-		downloadDataForAllStateHandles(miscFiles, dest, restoringThreadNum, closeableRegistry);
+		downloadDataForAllStateHandles(sstFiles, dest, closeableRegistry);
```

Comment:
could `closeableRegistry` be shared the same way as `executorService` in `RocksDBStateDataTransfer`? Then we do not need to pass to all class methods.